<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Shaun D Jackman">
  <meta name="dcterms.date" content="2016-06-06">
  <title>Assembling the Genome in a Bottle sequencing data with ABySS</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="reveal.js/css/theme/sky.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
  <h1 class="title">Assembling the Genome in a Bottle sequencing data with ABySS</h1>
  <h2 class="author">Shaun D Jackman</h2>
  <h3 class="date">2016-06-06</h3>
</section>

<section id="genome-in-a-bottle" class="slide level2">
<h1>Genome in a Bottle</h1>
<p>Assembling the Genome in a Bottle sequencing data with ABySS</p>
<p>Shaun Jackman <a href="http://twitter.com/sjackman">@sjackman</a></p>
<p>2016-06-06</p>
<p><a href="http://creativecommons.org/licenses/by/4.0/"><img src="images/cc-by.png" alt="Creative Commons Attribution License" /></a></p>
<p><a href="https://github.com/sjackman/giab-slides">Fork me on GitHub!</a></p>
</section>
<section id="shaun-jackman" class="slide level2">
<h1>Shaun Jackman</h1>
<p><a href="http://bcgsc.ca">BC Cancer Agency Genome Sciences Centre</a><br />
Vancouver, Canada<br />
<a href="http://twitter.com/sjackman">@sjackman</a> | <a href="https://github.com/sjackman">github.com/sjackman</a> | <a href="http://sjackman.ca">sjackman.ca</a></p>
<figure>
<img src="images/sjackman.jpg" />
</figure>
</section>
<section><section id="genome-in-a-bottle-data" class="titleslide slide level1"><h1>Genome in a Bottle Data</h1></section><section id="giab-data" class="slide level2">
<h1><a href="https://github.com/genome-in-a-bottle/giab_data_indexes">GIAB Data</a></h1>
<ul>
<li>7 individuals
<ul>
<li>Pilot individual (NA12878)</li>
<li>Ashkenazim Trio</li>
<li>Chinese Trio</li>
</ul></li>
<li>13 sequencing technologies
<ul>
<li>Illumina</li>
<li>Ion Torrent</li>
<li>10x Genomics</li>
<li>Pacbio</li>
<li>Oxford Nanopore</li>
<li>SOLiD</li>
<li>Complete Genomics</li>
<li>BioNano Genomics</li>
</ul></li>
</ul>
</section><section id="sequencing" class="slide level2">
<h1>Sequencing</h1>
<ul>
<li>Illumina 2x150 paired-end of 600 bp</li>
<li>Illumina 2x250 paired-end of 600 bp</li>
<li>Illumina 6 kbp mate-pair</li>
<li>Illumina Whole Exome</li>
<li>Ion Proton Exome</li>
<li>SOLiD</li>
<li>Moleculo</li>
<li>10x Genomics GemCode</li>
<li>PacBio</li>
<li>Oxford Nanopore</li>
<li>Complete Genomics</li>
<li>Complete Genomics LFR</li>
<li>BioNano Genomics Irys</li>
</ul>
</section></section>
<section><section id="assembly-pipeline" class="titleslide slide level1"><h1>Assembly Pipeline</h1></section><section id="assembly-pipeline-1" class="slide level2">
<h1>Assembly Pipeline</h1>
<ul>
<li>Inspect data</li>
<li>Trim adapters</li>
<li>Correct errors</li>
<li>Merge reads</li>
<li>Estimate <em>k</em>-mer abundance</li>
<li>Assemble</li>
<li>Scaffold</li>
<li>Close gaps</li>
<li>Polish</li>
<li>Align to reference</li>
<li>Calculate metrics</li>
<li>Count core genes</li>
<li>Generate reports</li>
</ul>
</section><section id="inspect-data" class="slide level2">
<h1>Inspect data</h1>
<ul>
<li>Check library and sequencing quality</li>
<li>Discard bad lanes</li>
<li>Trim bad cycles</li>
</ul>
<h3 id="tools">Tools</h3>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="http://multiqc.info">MultiQC</a></li>
</ul>
</section><section id="trim-adapters" class="slide level2">
<h1>Trim adapters</h1>
<p>Remove adapter sequence from reads</p>
<h3 id="tools-1">Tools</h3>
<ul>
<li><a href="https://github.com/marcelm/cutadapt">Cutadapt</a></li>
<li><a href="https://github.com/lh3/trimadap">Trimadap</a></li>
<li><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
<li><a href="https://github.com/sequencing/NxTrim">NxTrim</a> for Nextera mate-pair</li>
</ul>
</section><section id="correct-errors" class="slide level2">
<h1>Correct errors</h1>
<p>Reduce memory utilization and improve contiguity and correctness</p>
<h3 id="tools-2">Tools</h3>
<ul>
<li><a href="https://github.com/lh3/bfc">BFC</a></li>
</ul>
</section><section id="merge-reads" class="slide level2">
<h1>Merge reads</h1>
<ul>
<li>Merge overlapping reads</li>
<li>Fill the gap between paired-end reads</li>
</ul>
<h3 id="tools-3">Tools</h3>
<ul>
<li>ABySS-MergePairs</li>
<li><a href="http://www.bcgsc.ca/platform/bioinfo/software/konnector">Konnector</a></li>
</ul>
</section><section id="estimate-k-mer-abundance" class="slide level2">
<h1>Estimate <em>k</em>-mer abundance</h1>
<ul>
<li>Estimate memory usage prior to assembly</li>
<li>Pick a range of values for <em>k</em> to assemble</li>
</ul>
<h3 id="tools-4">Tools</h3>
<ul>
<li><a href="https://github.com/bcgsc/ntCard">ntCard</a></li>
<li><a href="https://github.com/pmelsted/KmerStream">KmerStreamer</a></li>
</ul>
</section><section id="assemble" class="slide level2">
<h1>Assemble</h1>
<p>Assemble reads into contigs</p>
<h3 id="tools-5">Tools</h3>
<ul>
<li><a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">ABySS</a></li>
<li><a href="https://www.broadinstitute.org/software/discovar/blog/">DISCOVARdenovo</a></li>
<li><a href="http://minia.genouest.org">Minia</a></li>
<li><a href="http://soap.genomics.org.cn/soapdenovo.html">SOAPdenovo</a></li>
</ul>
</section><section id="scaffold" class="slide level2">
<h1>Scaffold</h1>
<p>Join contigs into scaffolds</p>
<h3 id="tools-6">Tools</h3>
<ul>
<li><a href="http://www.bcgsc.ca/platform/bioinfo/software/links">LINKS</a> for mate-pair</li>
<li><a href="https://github.com/bcgsc/arcs">ARCS</a> for 10x Genomics</li>
</ul>
</section><section id="close-gaps" class="slide level2">
<h1>Close gaps</h1>
<ul>
<li>Fill gaps of Ns with sequence</li>
<li>Improve the contig N50</li>
</ul>
<h3 id="tools-7">Tools</h3>
<ul>
<li><a href="http://www.bcgsc.ca/platform/bioinfo/software/sealer">Sealer</a></li>
</ul>
</section><section id="polish" class="slide level2">
<h1>Polish</h1>
<p>Identify and remove errors from the assembly</p>
<h3 id="tools-8">Tools</h3>
<ul>
<li>Scrubber</li>
<li><a href="https://github.com/lh3/bfc">BFC</a> (off-label)</li>
<li><a href="http://www.broadinstitute.org/software/pilon/%5D">Pilon</a></li>
</ul>
</section><section id="align-to-reference" class="slide level2">
<h1>Align to reference</h1>
<p>Identify misassemblies</p>
<h3 id="tools-9">Tools</h3>
<ul>
<li><a href="https://github.com/lh3/bwa">BWA-MEM</a> <code>-xintractg</code></li>
<li><a href="http://mummer.sourceforge.net">MUMmer</a> for smaller genomes</li>
</ul>
</section><section id="calculate-metrics" class="slide level2">
<h1>Calculate metrics</h1>
<p>for contiguity, correctness, completeness</p>
<h3 id="tools-10">Tools</h3>
<ul>
<li><a href="http://quast.sourceforge.net/quast">QUAST</a> for smaller genomes</li>
<li><a href="https://github.com/bcgsc/abyss/blob/master/Misc/samtobreak.hs">ABySS-samtobreak</a></li>
<li><a href="http://lh3.github.io/2014/07/07/abreak-evaluating-de-novo-assemblies/">htsbox abreak</a> (no scaffold metrics)</li>
</ul>
</section><section id="count-core-genes" class="slide level2">
<h1>Count core genes</h1>
<p>Estimate assembly completeness without a reference</p>
<h3 id="tools-11">Tools</h3>
<ul>
<li><a href="http://busco.ezlab.org">BUSCO</a></li>
<li><a href="http://korflab.ucdavis.edu/datasets/cegma/">CEGMA</a></li>
</ul>
</section><section id="generate-reports" class="slide level2">
<h1>Generate reports</h1>
<p>Generate tables and figures of</p>
<p>contiguity, correctness, completeness</p>
<h3 id="tools-12">Tools</h3>
<ul>
<li><a href="http://quast.sourceforge.net/quast">QUAST</a></li>
<li><a href="http://rmarkdown.rstudio.com">RMarkdown</a></li>
</ul>
</section></section>
<section><section id="results" class="titleslide slide level1"><h1>Results</h1></section><section id="correct-errors-using-bfc" class="slide level2">
<h1>Correct errors using <a href="https://github.com/lh3/bfc">BFC</a></h1>
<ul>
<li>Reduce memory utilization from 975 GB to 418 Gb</li>
<li>Improve scaffold NG50 from 3.7 Mbp to 4.9 Mbp</li>
<li>Improve correctness (? no data)</li>
</ul>
</section><section id="contigs" class="slide level2">
<h1>Contigs</h1>
<figure>
<img src="images/contig-NG50.png" alt="Assemble contigs using ABySS" /><figcaption>Assemble contigs using <a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">ABySS</a></figcaption>
</figure>
</section><section id="scaffolds" class="slide level2">
<h1>Scaffolds</h1>
<figure>
<img src="images/scaffold-NG50.png" alt="Scaffold using ABySS" /><figcaption>Scaffold using <a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">ABySS</a></figcaption>
</figure>
</section><section id="links" class="slide level2">
<h1>LINKS</h1>
<p>Improve the scaffold</p>
<ul>
<li>N50 from 5.6 to 8.2 Mbp</li>
<li>NG50 from 4.6 to 6.8 Mbp</li>
<li>NGA50 from 4.4 to 5.3 Mbp</li>
</ul>
</section><section id="contiguity-vs-correctness" class="slide level2">
<h1>Contiguity vs correctness</h1>
<figure>
<img src="images/scaffold-NGA50-breakpoints.png" alt="Scaffold NGA50 vs breakpoints" /><figcaption>Scaffold NGA50 vs breakpoints</figcaption>
</figure>
</section></section>
<section><section id="fin" class="titleslide slide level1"><h1>fin</h1></section><section id="links-1" class="slide level2">
<h1>Links</h1>
<p><a href="https://github.com/bcgsc/abyss/blob/master/Misc/samtobreak.hs">ABySS-samtobreak</a> | <a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">ABySS</a> | <a href="https://github.com/bcgsc/arcs">ARCS</a> | <a href="https://github.com/lh3/bfc">BFC</a> | <a href="http://busco.ezlab.org">BUSCO</a> | <a href="https://github.com/lh3/bwa">BWA-MEM</a> | <a href="http://korflab.ucdavis.edu/datasets/cegma/">CEGMA</a> | <a href="https://github.com/marcelm/cutadapt">Cutadapt</a> | <a href="https://www.broadinstitute.org/software/discovar/blog/">DISCOVARdenovo</a> | <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> | <a href="http://lh3.github.io/2014/07/07/abreak-evaluating-de-novo-assemblies/">htsbox abreak</a> | <a href="https://github.com/pmelsted/KmerStream">KmerStreamer</a> | <a href="http://www.bcgsc.ca/platform/bioinfo/software/konnector">Konnector</a> | <a href="http://www.bcgsc.ca/platform/bioinfo/software/links">LINKS</a> | <a href="http://minia.genouest.org">Minia</a> | <a href="http://multiqc.info">MultiQC</a> | <a href="http://mummer.sourceforge.net">MUMmer</a> | <a href="https://github.com/bcgsc/ntCard">ntCard</a> | <a href="https://github.com/sequencing/NxTrim">NxTrim</a> | <a href="http://www.broadinstitute.org/software/pilon/%5D">Pilon</a> | <a href="http://quast.sourceforge.net/quast">QUAST</a> | <a href="http://rmarkdown.rstudio.com">RMarkdown</a> | <a href="http://www.bcgsc.ca/platform/bioinfo/software/sealer">Sealer</a> | <a href="http://soap.genomics.org.cn/soapdenovo.html">SOAPdenovo</a> | <a href="https://github.com/lh3/trimadap">Trimadap</a> | <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></p>
</section><section id="shaun-jackman-1" class="slide level2">
<h1>Shaun Jackman</h1>
<p><a href="http://bcgsc.ca">BC Cancer Agency Genome Sciences Centre</a><br />
Vancouver, Canada<br />
<a href="http://twitter.com/sjackman">@sjackman</a> | <a href="https://github.com/sjackman">github.com/sjackman</a> | <a href="http://sjackman.ca">sjackman.ca</a></p>
<figure>
<img src="images/sjackman.jpg" />
</figure>
</section></section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
